cores/nRF5/wiring_analog_nRF51.c

```diff
@@ -259,17 +246,25 @@ void analogWrite( uint32_t ulPin, uint32_t ulValue )
                                | ((uint32_t)GPIO_PIN_CNF_DRIVE_S0S1       << GPIO_PIN_CNF_DRIVE_Pos)
                                | ((uint32_t)GPIO_PIN_CNF_SENSE_Disabled   << GPIO_PIN_CNF_SENSE_Pos);
 
-      ulValue = mapResolution(ulValue, writeResolution, 8);
-
       pwmContext[i].value = ulValue;
 
       NRF_TIMER1->CC[pwmContext[i].channel] = ulValue;
 
       NRF_TIMER1->INTENSET = pwmContext[i].mask;
-
-      break;
+      pwmStatus[0].numActive++;
+      return;
     }
   }
+
+  // fallback to digitalWrite if no available PWM channel
+  if (ulValue < 128)
```

Comment:
Can the write resolution be higher than 8-bits? If so, we need to adjust this.

Reply:
On it. One thing i noticed is that the read and write resolution isnt bounded and for the NRF52 they are stored as signed integers so you could set the read or write resolution to +/-2e9!

If its alright ill add a static int with the half max value and a switch statement to set it in writeResolution so that valid values are used (and so they arent re-calculated on each analogWrite)

Reply:
For reference;
* nRF51 TIMER1 (the only timer currently used for PWM) is either 8b or 16b write resolution
* nRF52 is up to 32b write resolution for all 12 PWM channels